src/replica/Configuration.cc

```diff
@@ -172,6 +180,33 @@ unsigned int Configuration::databaseTransactionTimeoutSec() {
 }
 
 
+bool Configuration::schemaUpgradeWait() {
+    util::Lock const lock(_classMtx, _context(__func__));
+    return _schemaUpgradeWait;
+}
+
+
+void Configuration::setSchemaUpgradeWait(bool value) {
+    util::Lock const lock(_classMtx, _context(__func__));
+    _schemaUpgradeWait = value;
+}
+
+
+unsigned int Configuration::schemaUpgradeWaitTimeoutSec() {
+    util::Lock const lock(_classMtx, _context(__func__));
+    return _schemaUpgradeWaitTimeoutSec;
+}
+
+
+void Configuration::setSchemaUpgradeWaitTimeoutSec(unsigned int value) {
+    util::Lock const lock(_classMtx, _context(__func__));
+    if (0 == value) {
+        throw invalid_argument("Configuration::" + string(__func__) + "  0 is not allowed.");
+    }
```

Comment:
I suppose you could get rid of the option to wait or not, and just use "upgrade wait time"==0 to mean "don't wait for upgrade? Seems a little simpler, but either way seems ok to me.

Reply:
I too had the very same idea before I realized that I would prefer keeping knobs for "enabling something" and "setting a timeout for waiting for some event to happen" semantically separate both in the interface and the dependent code.